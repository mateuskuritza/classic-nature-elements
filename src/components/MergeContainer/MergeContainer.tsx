import styled from "styled-components";
import Button from "../Button/Button";

export default function MergeContainer({ merge }: { merge: CallableFunction }) {
	return (
		<Container>
			<img src="./assets/yin-yang.png" alt="yin yang merge" />
			<Button onClick={() => merge()}>Fundir!</Button>
		</Container>
	);
}

const Container = styled.div`
	img {
		object-fit: contain;
		width: 200px;
		height: 200px;
		margin-bottom: 10px;
	}
`;
