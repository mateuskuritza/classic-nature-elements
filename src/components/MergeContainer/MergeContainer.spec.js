import React from "react";
import { fireEvent, render, screen } from "@testing-library/react";
import MergeContainer from "./MergeContainer";

it("Should render MergeContainer", () => {
    const { container } = render(<MergeContainer />);
    expect(container).toMatchSnapshot();
});

it("Should render MergeContainer image", () => {
    const { container } = render(<MergeContainer />);
    const img = screen.getByRole("img");
    expect(img.getAttribute('src')).toBe('./assets/yin-yang.png');
    expect(container).toMatchSnapshot();
});

it("Should render MergeContainer button", () => {
    const { container } = render(<MergeContainer />);
    const button = screen.getByText('Fundir!');
    expect(button).toBeDefined();
    expect(container).toMatchSnapshot();
});

it("Should call merge function if click in MergeContainer button", () => {
    let value = false;
    const merge = () => value = true;
    const { container } = render(<MergeContainer merge={merge} />);
    const button = screen.getByText('Fundir!');
    expect(button).toBeDefined();

    expect(value).toBe(false);
    fireEvent.click(button);
    expect(value).toBe(true);

    expect(container).toMatchSnapshot();
});