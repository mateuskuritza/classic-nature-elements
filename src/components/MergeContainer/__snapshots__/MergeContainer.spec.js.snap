// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should call merge function if click in MergeContainer button 1`] = `
<div>
  <div
    class="sc-gsDKAQ hsklco"
  >
    <img
      alt="yin yang merge"
      src="./assets/yin-yang.png"
    />
    <button
      class="sc-bdvvtL LSrWP"
    >
      Fundir!
    </button>
  </div>
</div>
`;

exports[`Should render MergeContainer 1`] = `
<div>
  <div
    class="sc-gsDKAQ hsklco"
  >
    <img
      alt="yin yang merge"
      src="./assets/yin-yang.png"
    />
    <button
      class="sc-bdvvtL LSrWP"
    >
      Fundir!
    </button>
  </div>
</div>
`;

exports[`Should render MergeContainer button 1`] = `
<div>
  <div
    class="sc-gsDKAQ hsklco"
  >
    <img
      alt="yin yang merge"
      src="./assets/yin-yang.png"
    />
    <button
      class="sc-bdvvtL LSrWP"
    >
      Fundir!
    </button>
  </div>
</div>
`;

exports[`Should render MergeContainer image 1`] = `
<div>
  <div
    class="sc-gsDKAQ hsklco"
  >
    <img
      alt="yin yang merge"
      src="./assets/yin-yang.png"
    />
    <button
      class="sc-bdvvtL LSrWP"
    >
      Fundir!
    </button>
  </div>
</div>
`;

exports[`Should render modal if click in MergeContainer button 1`] = `
<div>
  <div
    class="sc-gsDKAQ hsklco"
  >
    <img
      alt="yin yang merge"
      src="./assets/yin-yang.png"
    />
    <button
      class="sc-bdvvtL LSrWP"
    >
      Fundir!
    </button>
  </div>
</div>
`;
